### PR TITLE
Fix Solr create error when sort geo is out of bounds

### PR DESCRIPTION
**Pull request by pklinef**
#### What does this PR do?
Fixes a Solr create error when the generated centroid of the location used for sorting is out of spatial bounds.

#### Who is reviewing it? 

@derekwilhelm 
@lamhuy 

#### Select relevant component teams: 

@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining
@millerw8

#### How should this be tested?

Verify a metacard with a location like `LINESTRING (-180 -63.0065, -180 -90, -180 -63.0065)` is successfully created. 

#### What are the relevant tickets?
Fixes: n/a

#### Checklist:
- [ n/a ] Documentation Updated
- [ n/a ] Update / Add Threat Dragon models
- [x] Update / Add Unit Tests
- [ n/a ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by pklinef** [comment]
build now

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java [hunk 1]
```diff
@@ -430,7 +431,31 @@ private String createCenterPoint(List<Serializable> values) {
...
+  }
+
+  /**
+   * Normalizes point for coordinate boundary wrapping. Truncates coordinates to a reasonable
+   * precision.
+   *
+   * @return Solr formatted coordinate
+   */
+  private String normalize(Point center) {
+    return DistanceUtils.normLatDEG(truncate(center.getY()))
```
Comment (millerw8) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java:444]: interesting function... considers going past the top of the world as if you turned back around and walked down.

Comment (pklinef) [reply catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java:444]: I realized I was starting to write a crappy version of it and then remembered Spatial4j would have the function I needed.  Thankfully it was public and already available to the Solr provider classes.

**Comment by bdeining** [comment]
hero looks good, linestring had a valid center point and point made it into solr 👍

**Comment by pklinef** [comment]
build now

**Pull request merged**